htmltest/check-link.go

```diff
@@ -347,6 +348,70 @@ func (hT *HTMLTest) checkMailto(ref *htmldoc.Reference) {
 		})
 		return
 	}
+
+	// split off domain, check mx, fallback to A or AAAA if that fails
+	var dnserr *net.DNSError
+	var ok bool
+
+	domain := strings.Split(ref.URL.Opaque, "@")[1]
+
+	for domain != "" {
```

Comment:
Bit of description here: we're going to loop until we find a valid record or error